src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/Termination/Checker.hs

```diff
@@ -100,48 +102,48 @@ checkTerminationShallow' topModule = do
           | Nothing <- order -> TerminatingFailed
 
 scanModule ::
-  (Members '[State CallMap] r) =>
+  (Members '[State CallMap, Reader SizeInfoMap] r) =>
```

Comment:
why add the Reader here? The goal should be to add the `Reader SizeInfoMap` only to places where it can be nonempty